### PR TITLE
Allow for customization of hintText style in InputField

### PR DESCRIPTION
**Pull request by dvdwasibi**
Since InputField is an un-opinionated (in terms of decoration) input field, it would be useful to be able to specify the hintText style as well.

Addresses #7850

### CODE REVIEW

File: packages/flutter/lib/src/material/input.dart [hunk 1]
```diff
@@ -47,6 +47,7 @@ class InputField extends StatefulWidget {
     this.keyboardType: TextInputType.text,
     this.hintText,
     this.style,
+    this.hintStyle,
```
Comment (abarth) [inline packages/flutter/lib/src/material/input.dart]: We should probably plumb this out through all the various text input widgets.  Also, we need a test.  :)

Comment (dvdwasibi) [inline packages/flutter/lib/src/material/input.dart]: Sounds good.

I was going to add them to the other Text widgets, but technically they are based on "Material Design Specs". Is it fine to provide this override?

Comment (Hixie) [inline packages/flutter/lib/src/material/input.dart]: Yes.

Comment (dvdwasibi) [inline packages/flutter/lib/src/material/input.dart]: The hintStyle/labelStyle seem highly opinionated for the InputContainer (which is used as a wrapper for the Input widget as well).  They are tied to the theme and the Dark/Lightness of the theme. 

After discussing offline, it seems best to keep the hintStyle override to the InputField, which is the non-material, open-ended widget for text input.

Tests added :)

**Comment by dvdwasibi** [comment]
@Hixie @abarth PTAL

Updated PR, after talking with Ian we decided to keep the scope of the hintText style override to just the InputField, since the Material-opinionated widgets (InputContainer, Input) have values that derived from the theme and the Dark/Lightness of theme.

**Comment by abarth** [comment]
LGTM

**Comment by dvdwasibi** [comment]
@Hixie @abarth, can you merge this in. I don't have permission :)

**Comment by abarth** [comment]
done!

**Pull request merged**